cpp/src/randomforest/randomforest.cu

```diff
@@ -268,40 +274,62 @@ void print_rf_detailed(const RandomForestMetaData<T, L>* forest) {
 template <class T, class L>
 void build_treelite_forest(ModelHandle* model,
                            const RandomForestMetaData<T, L>* forest,
-                           int num_features, int task_category) {
-  // Non-zero value here for random forest models.
-  // The value should be set to 0 if the model is gradient boosted trees.
-  int random_forest_flag = 1;
-  ModelBuilderHandle model_builder;
-  // num_output_group is 1 for binary classification and regression
-  // num_output_group is #class for multiclass classification which is the same as task_category
-  int num_output_group = task_category > 2 ? task_category : 1;
-  TREELITE_CHECK(TreeliteCreateModelBuilder(
-    num_features, num_output_group, random_forest_flag, &model_builder));
-
-  if (task_category > 2) {
-    // Multi-class classification
-    TREELITE_CHECK(TreeliteModelBuilderSetModelParam(
-      model_builder, "pred_transform", "max_index"));
+                           int num_features, int task_category,
+                           const char* filename,
+                           std::vector<unsigned char> data) {
+  bool check_val = data.empty();
+  if (not check_val) {
+    std::ofstream file("filename", std::ios::binary);
+    file.write((char*)&data[0], data.size());
```

Comment:
If I'm not mistaken this file write creates a file called "filename" that never gets deleted, which is an issue that might cause spurious files for users, and already causes it when running the pytests, see: 
```bash
(ns1106) ➜  python git:(fea-rf-pickling) ✗ pytest cuml/test/test_pickle.py
======================================== test session starts ========================================
platform linux -- Python 3.7.5, pytest-5.2.2, py-1.8.0, pluggy-0.13.0
rootdir: /home/dante/rapids/cuml-prs/1322/python, inifile: pytest.ini
collected 65 items

cuml/test/test_pickle.py ..........ssssssss....ssss..ssxxxxssss..XXXXssss..ss..ssxx..ss...    [100%]

========================================= warnings summary ==========================================
cuml/test/test_pickle.py:66
  /home/dante/rapids/cuml-prs/1322/python/cuml/test/test_pickle.py:66: UserWarning: Parameter should_downcast is deprecated, use convert_dtype in fit, fit_transform and transform  methods instead.
    UMAP=cuml.UMAP()

cuml/test/test_pickle.py::test_umap_pickle[model0-float32]
cuml/test/test_pickle.py::test_umap_pickle[model0-float64]
  /home/dante/rapids/cuml-prs/1322/python/cuml/test/test_pickle.py:232: UserWarning: Parameter should_downcast is deprecated, use convert_dtype in fit, fit_transform and transform  methods instead.
    cu_before_pickle_transform = model.fit_transform(X_train)

cuml/test/test_pickle.py::test_umap_pickle[model0-float32]
cuml/test/test_pickle.py::test_umap_pickle[model0-float64]
  /home/dante/rapids/cuml-prs/1322/python/cuml/test/test_pickle.py:241: UserWarning: Parameter should_downcast is deprecated, use convert_dtype in fit, fit_transform and transform  methods instead.
    cu_after_pickle_transform = cu_after_pickle_model.transform(X_train)

-- Docs: https://docs.pytest.org/en/latest/warnings.html
================= 27 passed, 28 skipped, 6 xfailed, 4 xpassed, 5 warnings in 10.29s =================
(ns1106) ➜  python git:(fea-rf-pickling) ✗ git status
On branch fea-rf-pickling
Your branch is up to date with 'origin/fea-rf-pickling'.

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	filename
	record.txt

nothing added to commit but untracked files present (use "git add" to track)
```

If this file is intended, the location might be wrong, or we might need a mechanism to handle this file

Reply:
the file was not intended, I updated the code so that we do not pass any file names from cython to c++, we do all of this in c++ directly.